pull_infra.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import argparse
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+
+
+def run(cmd, cwd=None, check=True):
+    print(f"🔧 Running: {' '.join(cmd)}")
+    result = subprocess.run(cmd, cwd=cwd, capture_output=True, text=True)
+    if check and result.returncode != 0:
+        print(f"❌ Error: {result.stderr.strip()}")
+        sys.exit(result.returncode)
```

Comment:
```suggestion
    result = subprocess.run(cmd, cwd=cwd, capture_output=True, check=check, text=True)
```

`subprocess.run` has a `check` feature that can save some complexity here. If you don't want to use it for some reason, I'd appreciate a comment to explain why not.

Reply:
It is probably better that way.